crates/wasi-provider/src/wasi_runtime.rs

```diff
@@ -71,64 +88,246 @@ impl WasiRuntime {
+                            i.name(),
+                            i.module()
+                        ),
+                    }
+                })
+                .collect::<Result<Vec<_>, _>>()
+            {
+                // We can't map errors here or it moves the send channel, so we
+                // do it in a match
+                Ok(m) => Ok(m),
+                Err(e) => {
+                    let message = "unable to load module";
+                    error!("{}: {:?}", message, e);
+                    status_sender
+                        .broadcast(ContainerState {
+                            terminated: Some(ContainerStateTerminated {
+                                message: Some(message.into()),
+                                reason: None,
+                                exit_code: 1,
+                                finished_at: Some(Time(chrono::Utc::now())),
+                                ..Default::default()
+                            }),
+                            ..Default::default()
+                        })
+                        .expect("status should be able to send");
+                    Err(e)
+                }
+            }?;
+
+            let instance = match wasmtime::Instance::new(&module, &imports) {
+                // We can't map errors here or it moves the send channel, so we
+                // do it in a match
+                Ok(m) => Ok(m),
+                Err(e) => {
+                    let message = "unable to instantiate module";
+                    error!("{}: {:?}", message, e);
+                    status_sender
+                        .broadcast(ContainerState {
+                            terminated: Some(ContainerStateTerminated {
+                                message: Some(message.into()),
+                                reason: None,
+                                exit_code: 1,
+                                finished_at: Some(Time(chrono::Utc::now())),
+                                ..Default::default()
+                            }),
+                            ..Default::default()
+                        })
+                        .expect("status should be able to send");
+                    // Converting from anyhow
+                    Err(format_err!("{}: {}", message, e))
+                }
+            }?;
+
+            // NOTE(taylor): In the future, if we want to pass args directly, we'll
+            // need to do a bit more to pass them in here.
+            info!("starting run of module");
+            status_sender
+                .broadcast(ContainerState {
+                    running: Some(ContainerStateRunning {
+                        started_at: Some(Time(chrono::Utc::now())),
+                    }),
+                    ..Default::default()
+                })
+                .expect("status should be able to send");
+            match instance
+                .get_export("_start")
+                .expect("export")
```

Comment:
Should this `expect` message be a bit more descriptive?

Reply:
Updated it with something clearer